examples/myapp-02 'cargo build' fails

Running `cargo build` in `examples/myapp-02` fails:
```
   Compiling myapp v0.1.0 (/home/dmitris
![Screenshot 2022-01-14 at 13 18 35](https://user-images.githubusercontent.com/31205/149514426-0c779154-177b-4f14-baf3-d3d5d1493c22.png)
![Screenshot 2022-01-14 at 13 18 35](https://user-images.githubusercontent.com/31205/149514485-be4ff808-d901-4ee0-8b87-5d73df53f785.png)
/dev/hack/gh/aya-rs/book/examples/myapp-02/myapp)
error[E0277]: the `?` operator can only be used on `Result`s, not `Option`s, in an async block that returns `Result`
  --> myapp/src/main.rs:31:49
   |
18 |   async fn main() -> Result<(), anyhow::Error> {
   |  ______________________________________________-
19 | |     let path = match env::args().nth(1) {
20 | |         Some(iface) => iface,
21 | |         None => panic!("not path provided"),
...  |
31 | |     let probe: &mut Xdp = bpf.program_mut("xdp")?.try_into()?;
   | |                                                 ^ use `.ok_or(...)?` to provide an error compatible with `Result<(), anyhow::Error>`
...  |
60 | |     Ok::<_, anyhow::Error>(())
61 | | }
   | |_- this function returns a `Result`
   |
   = help: the trait `FromResidual<Option<Infallible>>` is not implemented for `Result<(), anyhow::Error>`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `myapp` due to previous error
```

--- a/examples/myapp-01/myapp/src/main.rs
+++ b/examples/myapp-01/myapp/src/main.rs
@@ -1,9 +1,10 @@
 // ANCHOR: all
 // ANCHOR: use
+use anyhow::Context;
 use aya::{Bpf, include_bytes_aligned};
 use aya::programs::{Xdp, XdpFlags};
 use std::{
-    convert::{TryFrom,TryInto},
+    convert::TryInto,
     sync::Arc,
     sync::atomic::{AtomicBool, Ordering},
     thread,
@@ -41,7 +42,8 @@ fn try_main() -> Result<(), anyhow::Error> {
     ))?;
     let program: &mut Xdp = bpf.program_mut("myapp").unwrap().try_into()?;
     program.load()?;
-    program.attach(&opt.iface, XdpFlags::default())?;
+    program.attach(&opt.iface, XdpFlags::default())
+        .context("failed to attach the XDP program with default flags - try changing XdpFlags::default() to XdpFlags::SKB_MODE")?;
 
     let running = Arc::new(AtomicBool::new(true));
     let r = running.clone();

--- a/examples/myapp-02/README.md
+++ b/examples/myapp-02/README.md
@@ -24,5 +24,5 @@ cargo build
 ## Run
 
 ```bash
-cargo run --package myapp -bin myapp
+sudo ./target/debug/myapp ./target/bpfel-unknown-none/debug/myapp
 ```

--- a/examples/myapp-02/myapp/src/main.rs
+++ b/examples/myapp-02/myapp/src/main.rs
@@ -18,7 +18,7 @@ use myapp_common::PacketLog;
 async fn main() -> Result<(), anyhow::Error> {
     let path = match env::args().nth(1) {
         Some(iface) => iface,
-        None => panic!("not path provided"),
+        None => panic!("no path provided"),
     };
     let iface = match env::args().nth(2) {
         Some(iface) => iface,
@@ -28,7 +28,7 @@ async fn main() -> Result<(), anyhow::Error> {
     let data = fs::read(path)?;
     let mut bpf = Bpf::load(&data)?;
 
-    let probe: &mut Xdp = bpf.program_mut("xdp")?.try_into()?;
+    let probe: &mut Xdp = bpf.program_mut("xdp").ok_or(anyhow::anyhow!("failed to load bpf program"))?.try_into()?;
     probe.load()?;
     probe.attach(&iface, XdpFlags::default())?;
 

--- a/examples/myapp-02/myapp/src/main.rs
+++ b/examples/myapp-02/myapp/src/main.rs
@@ -1,3 +1,4 @@
+use anyhow::Context;
 use aya::{
     maps::perf::AsyncPerfEventArray,
     programs::{Xdp, XdpFlags},
@@ -28,9 +29,11 @@ async fn main() -> Result<(), anyhow::Error> {
     let data = fs::read(path)?;
     let mut bpf = Bpf::load(&data)?;
 
-    let probe: &mut Xdp = bpf.program_mut("xdp").ok_or(anyhow::anyhow!("failed to load bpf program"))?.try_into()?;
+    let probe: &mut Xdp = bpf.program_mut("xdp").unwrap().try_into()?;
     probe.load()?;
-    probe.attach(&iface, XdpFlags::default())?;
+    probe.attach(&iface, XdpFlags::default())
+        .context("failed to attach the XDP program with default flags - try changing XdpFlags::default() to XdpFlags::SKB_MODE")?;
+
 
     // ANCHOR: map
     let mut perf_array = AsyncPerfEventArray::try_from(bpf.map_mut("EVENTS")?)?;

--- a/examples/myapp-03/myapp/src/main.rs
+++ b/examples/myapp-03/myapp/src/main.rs
@@ -1,3 +1,4 @@
+use anyhow::Context;
 use aya::{
     maps::HashMap,
     maps::perf::{AsyncPerfEventArray},
@@ -26,9 +27,10 @@ async fn main() -> Result<(), anyhow::Error> {
     let data = fs::read(path)?;
     let mut bpf = Bpf::load(&data)?;
 
-    let probe: &mut Xdp = bpf.program_mut("xdp")?.try_into()?;
+    let probe: &mut Xdp = bpf.program_mut("xdp").unwrap().try_into()?;
     probe.load()?;
-    probe.attach(&iface, XdpFlags::default())?;
+    probe.attach(&iface, XdpFlags::default())
+        .context("failed to attach the XDP program with default flags - try changing XdpFlags::default() to XdpFlags::SKB_MODE")?;
 
     // ANCHOR: block_address
     let mut blocklist: HashMap<_, u32, u32> = HashMap::try_from(bpf.map_mut("BLOCKLIST")?)?;

